sdfg_utils.js

```diff
@@ -181,19 +323,18 @@ function parse_sdfg(sdfg_json) {
     return JSON.parse(sdfg_json, reviver);
 }
 
-function isDict(v) {
-    return typeof v === 'object' && v !== null && !(v instanceof Array) && !(v instanceof Date);
-}
+function replacer(name, val) {
```

Comment:
What was the rationale behind changing the replacer? Did it fail on circular object structures?

Reply:
Yes, the main reason was to prevent adding Edge objects to break the circular structure.